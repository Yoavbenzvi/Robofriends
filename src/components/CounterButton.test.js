//Stateful component testing tryout

import { shallow } from "enzyme";
import React from "react";
import CounterButton from "./CounterButton.js";

describe("CounterButton component testing", () => {
	it("shallow render", () => {
		const mockColor = "red";
		const wrapper = shallow(<CounterButton color={mockColor} />);
		expect(wrapper).toMatchSnapshot();
	});

	it("increment correctly", () => {
		const wrapper = shallow(<CounterButton />);

		wrapper.find("[id='counter']").simulate("click");
		expect(wrapper.state()).toEqual({ count: 2 });
	});

	it("test props", () => {
		const mockColor = "red";
		const wrapper = shallow(<CounterButton color={mockColor} />);

		expect(wrapper.props().color).toEqual("red");
	});
});
