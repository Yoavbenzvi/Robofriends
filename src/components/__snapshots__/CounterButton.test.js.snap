// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton component testing shallow render 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  1
</button>
`;
